### PR TITLE
Add support for NATURAL JOIN and JOIN USING

### PR DESCRIPTION
**Pull request by Tekki**
These are the changes for #59.

### CODE REVIEW

File: lib/SQL/Abstract/mysql.pm [hunk 1]
```diff
@@ -228,12 +239,20 @@ names, but also array references with tables to generate C<JOIN> clauses for.
   $abstract->select(['foo', ['bar', foo_id => 'id']]);
 
   # "select * from foo join bar on (foo.id = bar.foo_id)"
-  $abstract->select(['foo', ['bar', 'foo.id' => 'bar.foo_id']]);
+  $abstract->select(['foo', ['bar', foo.id => 'bar.foo_id']]);
```
Comment (jhthorsen) [inline lib/SQL/Abstract/mysql.pm]: I think you should keep the quotes :)

**Review by jhthorsen — APPROVED** [review]
I can’t really find any issues with this, so feel free to merge it when you want 👍

The only thing I was wondering is if we should add experimental support for just one argument, a table name, then automatically use table_id or id_table as key... but maybe it won’t be possible to come to an agreement which one is the preferred one? 😉

Btw: That comment above is out of scope, so it shouldn’t affect the time to merge.

File: lib/SQL/Abstract/mysql.pm (hunk 1 above)
Comment (Tekki) [reply lib/SQL/Abstract/mysql.pm]: I thought TMTOWTDIBSCINABTE 😀 
Edit: But of course it's not consistent.

**Comment by Tekki** [comment]
> The only thing I was wondering is if we should add experimental support for just one argument, a table name, then automatically use table_id or id_table as key... but maybe it won’t be possible to come to an agreement which one is the preferred one? 😉

Or interpret one argument as NATURAL JOIN? We can try such things, but only if all discussions with graduates in Defense Against the Dark Arts are automatically assigned to you.

**Comment by jhthorsen** [comment]
Btw: You can make a new release if you like.

**Pull request merged**